src/SonarQube.Client.Tests/Services/SonarQubeServiceTests.cs

```diff
@@ -337,51 +337,42 @@ public void EnsureIsConnected_WhenNotConnected_ShouldThrow()
         }
 
         [TestMethod]
-        public async Task ThrowWhenNotConnected()
+        public void ThrowWhenNotConnected()
         {
             // Arrange
             var client = new Mock<ISonarQubeClient>();
             var sqService = new SonarQubeService(WrapInMockFactory(client));
 
             // Act & Assert
-            await AssertExceptionThrownWhenNotConnected(() =>
+            AssertExceptionThrownWhenNotConnected(() =>
                 sqService.GetAllOrganizationsAsync(CancellationToken.None));
 
-            await AssertExceptionThrownWhenNotConnected(() =>
+            AssertExceptionThrownWhenNotConnected(() =>
                 sqService.GetAllPluginsAsync(CancellationToken.None));
 
-            await AssertExceptionThrownWhenNotConnected(() =>
+            AssertExceptionThrownWhenNotConnected(() =>
                 sqService.GetAllProjectsAsync("organizationKey", CancellationToken.None));
 
-            await AssertExceptionThrownWhenNotConnected(() =>
+            AssertExceptionThrownWhenNotConnected(() =>
                 sqService.GetAllPropertiesAsync(CancellationToken.None));
 
-            await AssertExceptionThrownWhenNotConnected(() =>
+            AssertExceptionThrownWhenNotConnected(() =>
             {
                 sqService.GetProjectDashboardUrl("projectKey");
                 return Task.Delay(0);
             });
 
-            await AssertExceptionThrownWhenNotConnected(() =>
+            AssertExceptionThrownWhenNotConnected(() =>
                 sqService.GetQualityProfileAsync("projectKey", SonarQubeLanguage.CSharp, CancellationToken.None));
 
-            await AssertExceptionThrownWhenNotConnected(() =>
+            AssertExceptionThrownWhenNotConnected(() =>
                 sqService.GetRoslynExportProfileAsync("qualityProfileName", SonarQubeLanguage.CSharp, CancellationToken.None));
         }
 
-        private async Task AssertExceptionThrownWhenNotConnected(Func<Task> action)
+        private void AssertExceptionThrownWhenNotConnected(Func<Task> action)
```

Comment:
Rename the constructor argument from `action` to something else because this is a `Func<Task>`.